### ISSUE TITLE
design new auth login with blockstack auth button

### ISSUE THREAD

**Issue by guylepage3** [blockstack/designs#204]
Design a new Blockstack Auth button for developers to use in building their applications.

**Requirements**
- Design must be;
  - Neutral
  - Consistent with Blocstack
  - Recognizable as a Blockstack product
  - Scalable to work on all devices
  - Simple

Requested by `@drob (Robert)` on Blockstack Slack `#onename-support` channel.

>I’m creating an android app which hopes to use blockstack for auth, do you guys have any suggestions around branding for One Name or blockstack (ie. do you have a “Sign In with Blockstack” button or image?)

**Examples**
- [Facebook Login Button](https://developers.facebook.com/docs/facebook-login/web/login-button)
- [Login with Twitter](https://www.google.com/search?q=login+with+twitter+button&newwindow=1&rlz=1C5CHFA_enUS701US701&espv=2&biw=2133&bih=1222&tbm=isch&tbo=u&source=univ&sa=X&ved=0ahUKEwiMtv-g_6HSAhVF1RQKHU0PDKwQsAQIQQ)
- [Facebook](https://developers.facebook.com/docs/plugins/follow-button)
- [Github](https://buttons.github.io/)
- [Twitter](https://publish.twitter.com/#)
- [Google +1](https://developers.google.com/+/web/+1button/)

**Comment by guylepage3** [comment]
Pegging this issue to [#235: design hello blockstack app to be consistent with brand](https://github.com/blockstack/blockstack-portal/issues/235)

**Comment by guylepage3** [comment]
![btn-blockstack-id-1 2x](https://cloud.githubusercontent.com/assets/1711854/23188360/6924551c-f85c-11e6-860a-0ffc6a5be27d.png)
![btn-blockstack-id-2 2x](https://cloud.githubusercontent.com/assets/1711854/23188361/692b13fc-f85c-11e6-921b-6045af3d3b8a.png)

**Comment by jcnelson** [comment]
I like the former.  I don't like having `id` since (1) it's specific to English and Roman characters, and (2) it exposes an implementation detail of our system (i.e. the `.id` namespace, the `id` prefix in `OP_RETURN`s, etc.).

**Comment by pstan26** [comment]
@guylepage3   Really liking the first one since it is most brand-aligned and looks like uncovering a door's peephole. the second one looks great, but may be too similar to LinkedIn.

Will the boxes have rounded edges?

**Comment by shea256** [comment]
Nice @guylepage3!

Although, I feel like it'll be hard to fit into a lot of websites.

Will we be able to offer multiple versions?

If your site doesn't have flat design, for example, this wouldn't work when you integrate.

This is the reason there are so many available social login button styles:

![screen shot 2017-02-21 at 5 47 38 pm](https://cloud.githubusercontent.com/assets/994646/23188758/ead4eeea-f85d-11e6-84fc-015c7a8a8973.png)

**Comment by muneeb-ali** [comment]
I like the first version!

**Comment by guylepage3** [comment]
![btn-blockstack-id-3 2x](https://cloud.githubusercontent.com/assets/1711854/23188901/77c7c462-f85e-11e6-959e-b37e47636f82.png)

**Comment by guylepage3** [comment]
@shea256 i look at both facebook and twitter and they offer only flat design buttons
<img width="1133" alt="screenshot 2017-02-21 17 53 35" src="https://cloud.githubusercontent.com/assets/1711854/23188971/c15f5428-f85e-11e6-987a-ed8df84791f2.png">
.

**Comment by guylepage3** [comment]
<img width="786" alt="screenshot 2017-02-21 17 54 36" src="https://cloud.githubusercontent.com/assets/1711854/23188998/d8170a76-f85e-11e6-8300-9e992e8c751e.png">

**Comment by larrysalibra** [comment]
I like the one without `id` better.

@shea256 most of the different versions you shared were either created by 3rd parties or were previously offered. 

Usually offering buttons that can go with light/dark color schemes with a couple sizes is good enough. 

These buttons always look out of place: one could say that's the point because it makes the button easy to notice across different sites.

**Comment by shea256** [comment]
Ah good points @guylepage3 @larrysalibra.

Next question: if we put this button next to the twitter and facebook buttons, how would they look?

**Comment by guylepage3** [comment]
One thing that I'd love to do is get the wording down to just one word. **Login** or something like that. Thoughts?

**Comment by guylepage3** [comment]
>Next question: if we put this button next to the twitter and facebook buttons, how would they look?

Will do that now.. 😄

**Comment by larrysalibra** [comment]
> One thing that I'd love to do is get the wording down to just one word. Login or something like that. Thoughts?

Facebook has their F logo and the words "Log In":
<img width="121" alt="screen shot 2017-02-22 at 1 08 18 am" src="https://cloud.githubusercontent.com/assets/597182/23189400/6ce7df72-f89b-11e6-99b7-2a084b7a69aa.png">

Can we do that when our logo (the 3d dot) is relatively unknown?

**Comment by guylepage3** [comment]
Yeah. facebook has that one as well

**Comment by guylepage3** [comment]
Facebook also has a "best practices" page for when developers need to create their own button

https://developers.facebook.com/docs/facebook-login/best-practices

**Comment by guylepage3** [comment]
And this one is the one that comes with their SDK
![image](https://cloud.githubusercontent.com/assets/1711854/23189549/4f6276d6-f861-11e6-955d-75604c2a3142.png)

**Comment by guylepage3** [comment]
![artboard 2 2x](https://cloud.githubusercontent.com/assets/1711854/23190000/9732a6a0-f863-11e6-9f20-a910656ecb47.png)

**Comment by guylepage3** [comment]
![artboard 2 2x](https://cloud.githubusercontent.com/assets/1711854/23190063/dcba79f0-f863-11e6-844e-c682079faba2.png)

**Comment by guylepage3** [comment]
![artboard 2](https://cloud.githubusercontent.com/assets/1711854/23190070/e866efea-f863-11e6-93b4-953cc6142f70.png)

**Comment by guylepage3** [comment]
![blockstack-facebook-buttons](https://cloud.githubusercontent.com/assets/1711854/23190223/a923ad22-f864-11e6-9a89-b0699a53035e.png)

**Comment by guylepage3** [comment]
![blockstack-facebook-buttons 2x](https://cloud.githubusercontent.com/assets/1711854/23190241/c0eee5f2-f864-11e6-9d0b-93604096435e.png)

**Comment by shea256** [comment]
Much better! Really like these two.

**Comment by guylepage3** [comment]
closing issue

**Comment by guylepage3** [comment]
finalized all log in with blockstack buttons

<img width="369" alt="screenshot 2017-02-22 12 13 11" src="https://cloud.githubusercontent.com/assets/1711854/23223187/59769f84-f8f8-11e6-99b1-f1e54b11e438.png">

**Comment by shea256** [comment]
@guylepage3 Can we do one of the following:

- "Log in with Blockstack"
- "Sign in with Blockstack"

One typically sees a lowercase "in" with auth buttons.

![screen shot 2017-02-22 at 12 19 36 pm](https://cloud.githubusercontent.com/assets/994646/23223488/344fbbf4-f8f9-11e6-883e-4e117d6bcef8.png)

**Comment by guylepage3** [comment]
@shea256 why do "Sign in"? we discussed this with @john-light last year and decided that user's "Log in" and "Sign out"

**Comment by guylepage3** [comment]
closing this issue

**Comment by shea256** [comment]
> @shea256 why do "Sign in"? we discussed this with @john-light last year and decided that user's "Log in" and "Sign out"

Out of the 6 horsemen of the internet (FANGAM), only Facebook uses "log in" and the rest use "sign in":

- Facebook: "Log In"
- Apple: "Sign in"
- Netflix: "Sign In"
- Google: "Sign in"
- Amazon: "Sign in"
- Microsoft: "Sign in"

Bonus:

- Uber: "SIGN IN"

**Comment by guylepage3** [comment]
Whoa. I had not noticed this.. It definitely makes sense to go with **Sign In**. I don't know why we decided to go with Log In to begin with. Looks like we followed the two largest social networks;

Facebook - "Log In"
Twitter - "Log In"
<img width="280" alt="screenshot 2017-03-27 11 21 11" src="https://cloud.githubusercontent.com/assets/1711854/24364016/84837b2e-12df-11e7-8f7f-68dff90aaf09.png">

Having said that, it does make sense to use **"Sign In"** with Blockstack as the user is actually signing in with their keys. Either one works.
